core/src/main/java/org/elasticsearch/indices/recovery/RecoveryFileChunkRequest.java

```diff
@@ -105,11 +103,9 @@ public void readFrom(StreamInput in) throws IOException {
         String name = in.readString();
         position = in.readVLong();
         long length = in.readVLong();
-        String checksum = in.readOptionalString();
+        String checksum = in.readString();
         content = in.readBytesReference();
-        Version writtenBy = null;
-        String versionString = in.readOptionalString();
-        writtenBy = Lucene.parseVersionLenient(versionString, null);
+        Version writtenBy = Lucene.parseVersionLenient(in.readString(), null);
```

Comment:
should we just use `Version.parseLenient(in.readString())` it must be non-null right?